zebra/zebra_routemap.c

```diff
@@ -1473,6 +1502,80 @@ static void zebra_route_map_event(route_map_event_t event,
 	route_map_notify_dependencies(rmap_name, RMAP_EVENT_MATCH_ADDED);
 }
 
+/* The function checks if the changed routemap specified by parameter rmap
+ * matches the configured protocol routemaps in proto_rm table. If there is
+ * a match then rib_update_table() to process the routes.
+ */
+void zebra_rib_table_rm_update(char *rmap)
+{
+	int i = 0;
+	struct route_table *table;
+	char *rmap_name;
+
+	for (i = 0; i <= ZEBRA_ROUTE_MAX; i++) {
+		rmap_name = proto_rm[AFI_IP][i];
+		if (rmap_name && (strcmp(rmap_name, rmap) == 0)) {
+			if (IS_ZEBRA_DEBUG_EVENT)
+				zlog_debug("%s : AFI_IP rmap %s, route type %d",
+					   __func__, rmap, i);
```

Comment:
There is a conversion function to convert a ZEBRA_ROUTE_XXX to a useful string, let's use this.